### PR TITLE
N°5996 - During creation of csv file, in function lookup remove warning if initial field is empty

### PR DESCRIPTION
**Pull request by accognet**
During creation of csv file, in function lookup remove warning if initial field is empty.

Example :
During the collection process, in my initial data, oslicence_id is empty. I have a lookup on this field. This lookup does not find any data and it's normal. But in the logs, there is a warning :
[Warning]       No mapping found with key: '47_', 'oslicence_id' will be set to zero.

I want to remove this warning

### CODE REVIEW

**Comment by Hipska** [comment]
This change is not needed, you only need to set `$this->bIgnoreMappingErrors` to `true`.

**Comment by accognet** [comment]
You are right, if you set $this->bIgnoreMappingErrors to true, the warning goes away, but there is still a debug message. And for me it's not the same if initial field is empty or not.

**Comment by Hipska** [comment]
Ok, I understand. But then I would add the logic more to line 140 where the current message is emitted.

**Comment by accognet** [comment]
It's an option, but at the beginning of the function, this avoids looking for something that does not exist.

**Comment by odain-cbd** [comment]
it would be nice to add to cover lookup function with tests. nominal ones and one regarding the warning you are currently removing. that way our CI will also test this part of the code on different versions of PHP...

File: core/utils.class.inc.php [hunk 1]
```diff
@@ -213,9 +214,10 @@ private static function CreateEventIssue($sMessage)
 		$oClient->Create("EventIssue", $aFields, 'create event issue from collector $sCollectorName execution.');
 	}
 
-	static public function MockLog($oMockedLogger)
```
Comment (odain-cbd) [inline core/utils.class.inc.php:216]: test are broken. below signature would fix them i guess:
static public function MockLog($oMockedLogger, $iMockLogLevel=LOG_ERR)

Comment (accognet) [reply core/utils.class.inc.php:216]: Done

**Review by Molkobain — APPROVED** [review]

**Review by odain-cbd — APPROVED** [review]
just check it is fine for our ci as well before merging :)

great that you added so much tests. our coverage was really bad on this part of the code

**Pull request merged**